akka-actor/src/main/scala/akka/serialization/Serialization.scala

```diff
@@ -246,10 +246,17 @@ class Serialization(val system: ExtendedActorSystem) extends Extension {
   val serializerByIdentity: Map[Int, Serializer] =
     Map(NullSerializer.identifier -> NullSerializer) ++ serializers map { case (_, v) ⇒ (v.identifier, v) }
 
-  private def shouldWarnAboutJavaSerializer(serializedClass: Class[_], serializer: Serializer) =
+  private def shouldWarnAboutJavaSerializer(serializedClass: Class[_], serializer: Serializer) = {
+    def isJavaPrimitive(packageName: String): Boolean = {
+      val primitives = List("java.lang.Boolean", "java.lang.Byte", "java.lang.Character", "java.lang.Double", "java.lang.Float", "java.lang.Integer", "java.lang.Long", "java.lang.Short", "java.lang.String")
```

Comment:
- should be allocated once somewhere (in object most likely)
- should be a set for more efficient `contains`
- would we be able to obtain those from config? I think so, since system should be available here :) Then we'd allow people to silence the warnings if they "really know about one"